fft/unit_test/Test_Utils.hpp

```diff
@@ -60,10 +60,102 @@ void display(ViewType& a) {
   auto* data = h_a.data();
 
   std::cout << std::scientific << std::setprecision(16) << std::flush;
-  for (int i = 0; i < n; i++) {
+  for (std::size_t i = 0; i < n; i++) {
     std::cout << label + "[" << i << "]: " << i << ", " << data[i] << std::endl;
   }
   std::cout << std::resetiosflags(std::ios_base::floatfield);
 }
 
-#endif
\ No newline at end of file
+/// \brief Kokkos equivalent of fft1 with numpy
+/// def fft1(x):
+///    L = len(x)
+///    phase = -2j * np.pi * (np.arange(L) / L)
+///    phase = np.arange(L).reshape(-1, 1) * phase
+///    return np.sum(x*np.exp(phase), axis=1)
+///
+/// \tparam ViewType: Input and output view type
+///
+/// \param in [in]: Input rank 1 view
+/// \param out [out]: Output rank 1 view
+template <typename ViewType>
+void fft1(ViewType& in, ViewType& out) {
+  using value_type      = typename ViewType::non_const_value_type;
+  using real_value_type = KokkosFFT::Impl::base_floating_point_type<value_type>;
+
+  static_assert(KokkosFFT::Impl::is_complex_v<value_type>,
+                "fft1: ViewType must be complex");
+
+  const value_type I(0.0, 1.0);
+  std::size_t L = in.size();
+
+  Kokkos::parallel_for(
+      "KokkosFFT::Test::fft1",
+      Kokkos::TeamPolicy<execution_space>(L, Kokkos::AUTO),
+      KOKKOS_LAMBDA(
+          const Kokkos::TeamPolicy<execution_space>::member_type& team_member) {
+        const int j = team_member.league_rank();
+
+        value_type sum = 0;
+        Kokkos::parallel_reduce(
+            Kokkos::TeamThreadRange(team_member, L),
+            [&](const int i, value_type& lsum) {
+              auto phase = -2 * I * M_PI * static_cast<real_value_type>(i) /
```

Comment:
```suggestion
              auto phase = -2 * I * Kokkos::numbers::pi * static_cast<real_value_type>(i) /
```

If you support recent enough Kokkos ?

Reply:
Sure

Reply:
I got a compile error for nvcc

`/work/fft/unit_test/Test_Utils.hpp(102): error: identifier "Kokkos::numbers::pi" is undefined in device code`

Have you already succeeded to use this in DDC?